cloudinit/sources/__init__.py

```diff
@@ -349,6 +354,18 @@ def _unpickle(self, ci_pkl_version: int) -> None:
                 raise DatasourceUnpickleUserDataError() from e
         if not hasattr(self, "extra_hotplug_udev_rules"):
             self.extra_hotplug_udev_rules = None
+        if not hasattr(self, "metadata_address"):
+            self.metadata_address = None
```

Comment:
In this function we have so many of the boilerplate setup of instance attributes to the same default values. Would it make sense to loop over expected_attrs and set the expect defaults to avoid the if not hasattr duplication?
```diff
--- a/cloudinit/sources/__init__.py
+++ b/cloudinit/sources/__init__.py
@@ -331,12 +331,26 @@ class DataSource(CloudInitPickleMixin, metaclass=abc.ABCMeta):
 
     def _unpickle(self, ci_pkl_version: int) -> None:
         """Perform deserialization fixes for Paths."""
-        if not hasattr(self, "vendordata2"):
-            self.vendordata2 = None
-        if not hasattr(self, "vendordata2_raw"):
-            self.vendordata2_raw = None
-        if not hasattr(self, "skip_hotplug_detect"):
-            self.skip_hotplug_detect = False
+        expected_attrs = [
+            "_crawled_metadata",
+            "_platform_type",
+            "_subplatform",
+            "ec2_metadata",
+            "extra_hotplug_udev_rules",
+            "metadata_address",
+            "network_json",
+            "skip_hotplug_detect",
+            "vendordata2",
+            "vendordata2_raw",
+        ]
+        value_defaults = {
+            "skip_hotplug_detect": False,
+            "ec2_metadata": UNSET,
+            "network_json": UNSET,
+        }
+        for attr in expected_attrs:
+            if not hasattr(self, attr):
+                setattr(self, attr, value_defaults.get(attr))
         if hasattr(self, "userdata") and self.userdata is not None:
             # If userdata stores MIME data, on < python3.6 it will be
             # missing the 'policy' attribute that exists on >=python3.6.
@@ -352,20 +366,6 @@ class DataSource(CloudInitPickleMixin, metaclass=abc.ABCMeta):
                     e,
                 )
                 raise DatasourceUnpickleUserDataError() from e
-        if not hasattr(self, "extra_hotplug_udev_rules"):
-            self.extra_hotplug_udev_rules = None
-        if not hasattr(self, "metadata_address"):
-            self.metadata_address = None
-        if not hasattr(self, "_platform_type"):
-            self._platform_type = None
-        if not hasattr(self, "_subplatform"):
-            self._subplatform = None
-        if not hasattr(self, "_crawled_metadata"):
-            self._crawled_metadata = None
-        if not hasattr(self, "network_json"):
-            self.network_json = UNSET
-        if not hasattr(self, "ec2_metadata"):
-            self.ec2_metadata = UNSET
 
     def __str__(self):
         return type_utils.obj_name(self)
```